NavigationBarAssignment/NavigationBarAssignment/AddProductViewController.m

```diff
@@ -0,0 +1,121 @@
+ */
+-(BOOL)isWhitespace:(NSString*)name{
+    NSString *trimmedName=[name stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+    if(trimmedName.length==0){
+        return NO;
+    }
+    return YES;
+}
+
+/** Checks for alphabets and other characters in the string
+ * \param input The input whose content character set is to be checked
+ * \returns Returns yes if alphabets are only present else it returns no
+ */
+-(BOOL)isAlphaOnly:(NSString *)input
+{
+    NSString *regExp = @"[a-zA-Z]+";
+    NSPredicate *regexTest = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", regExp];
+    return [regexTest evaluateWithObject:input];
+}
+
+/** Checks for numbers, decimal point and number of digits
+ * \param input The input whose content number set is to be checked
+ * \returns Returns yes if numbers, decimal point and number of digits within limit are only present else it returns no
+ */
+-(BOOL)isNumericOnly:(NSString *)input
+{
+    NSString *regExp = @"[0-9]{2,5}[.]{0,1}[0-9]{0,2}";
+    NSPredicate *regexTest = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", regExp];
+    return [regexTest evaluateWithObject:input];
+}
+/** Clears all text fields
+ */
+-(void)clearTextField{
+    textFieldProductName.text=@"";
+    textFieldProductPrice.text=@"";
+    textFieldProductCategory.text=@"";
+}
+/** Saves product to the array
+ */
+-(void)saveProduct{
+    Products *product=[[Products alloc]init];
+    product.productName=textFieldProductName.text;
+    product.productPrice=[textFieldProductPrice.text floatValue];
+    product.productCategory=textFieldProductCategory.text;
+    AppDelegate *appDelegate = (AppDelegate*)[[UIApplication sharedApplication] delegate];
+    [appDelegate.arrayProducts addObject:product];
+}
+/** Pops previous view controller
+ */
+-(void)popViewController{
+    [self.navigationController popViewControllerAnimated:YES];
+}
+/** Validates textfields and saves that to array on click of button save
+ * \param sender The id of save button
+ * \returns Returns action on click of button
+ */
+- (IBAction)saveButtonTapped:(id)sender {
+    NSString *productName=textFieldProductName.text;
+    NSString *productCategory=textFieldProductCategory.text;
+    if([self isAlphaOnly:productName]==NO || [self isAlphaOnly:productCategory]== NO){
+        NSLog(@"Only character set allowed.");
+    }
+    if([self isNumericOnly:textFieldProductPrice.text]==NO ){
+        NSLog(@"Number ranging from 10 to 99999 and after decimal point only two numbers.");
+    }
+    if([self isWhitespace:productName]== NO){
+        [textFieldProductName becomeFirstResponder];
+    }
+    else if([self isWhitespace:textFieldProductPrice.text]== NO){
+        [textFieldProductPrice becomeFirstResponder];
+    }
+    else if([self isWhitespace:productCategory]== NO){
+        [textFieldProductCategory becomeFirstResponder];
+    }
+    else{
+        [self saveProduct];
+        [self popViewController];
+    }
+}
```

Comment:
- no need to check boolean values explicitly for bool returning type method == NO
- call validation methods priority wise in order, and return method exactly where first validation fails.
- use if else for all, if you want to validate one by one.